packageurl.go

```diff
@@ -251,6 +251,17 @@ func FromString(purl string) (PackageURL, error) {
 		remainder = remainder[:index]
 	}
 
+	atIndex := strings.Index(remainder, "@")
```

Comment:
Perhaps add a guiding comment
```suggestion
      // start out by extracting the purl version (if one is specified) to prevent it from being type-adjusted (might change casing, for example)
	atIndex := strings.Index(remainder, "@")
```

Reply:
Added 8129be64d5bba583d7f82091d41d735acd80b89e